libsolidity/ast/AsmJsonImporter.cpp

```diff
@@ -178,10 +180,7 @@ yul::Literal AsmJsonImporter::createLiteral(Json::Value const& _node)
 		);
 	}
 	else if (kind == "string")
-	{
 		lit.kind = yul::LiteralKind::String;
-		astAssert(scanner.currentToken() == Token::StringLiteral, "Expected string literal!");
```

Comment:
The problem here was that ``lit.value.str()`` merely contains the *contents* of the string literal, but not the enclosing delimiters, so in our test case the scanner would interpret "abc" as an identifier, not as a string literal.

I'm not entirely sure, if we can just blindly accept the value here instead or whether there should be some transformation or some kind of validation. I'd reckon the string could pretty much contain anything, right? Any thoughts?

Reply:
I'll locally investigate a bit more what happens with weird kinds of strings and escapes, etc.

Reply:
I haven't found anything wrong with it - things like ``"\n"`` or ``'\n"'`` or ``"\x00\x01\x02"`` will all be properly escaped in the export and then properly un-escaped in the ast import. And since we can have any hex byte in the string, it truly can be anything, so there's no point in validating anything here.

Reply:
Ah, the length maybe... yeah, I'll require the length to be <= 32 - we do already check that during parsing, so we need it here, then it should be fine.

Reply:
[done that]

Reply:
String literals must have length <= 32? Didn't know that

Reply:
In yul they do - in solidity they don't. In yul they have to fit into one variable/stack slot, so they have to fit into 256 bits.

Reply:
For more than 32 bytes what would ``let a := "some...long...string"`` mean? That's why.